lib/common.js

```diff
@@ -443,6 +443,18 @@ exports.readMysqlValue = function (
       } else if (defPrefix === 'varbin') {
         result = parser.parseBuffer(size);
       } else {
+        if (column.charset !== null) {
+          switch (column.charset) {
+            // Javascript UTF8 always allows up to 4 bytes per character
+            case 'utf8mb3':
+            case 'utf8mb4':
+              column.charset = 'utf8';
+              break;
+          }
+
+          parser._encoding = column.charset;
```

Comment:
I don't think this breaks anything, but I'm worried something else might depend on the `_encoding` being set to something else, and this is only really supposed to change it for the column being read. Can we call `parseBuffer(size).toString(column.charset ?? parser._encoding)` instead?